brightness-mode-explainer.md

```diff
@@ -0,0 +1,129 @@
+  - [\<video\> integration to allow e.g. granular brightness control](https://github.com/w3c/screen-wake-lock/issues/129#issuecomment-926603108) is an instance of the above. This is most likely better handled elsewhere and should probably be a UA-specific control.
+
+## Proposed Solutions
+
+The following represent some rough proposals that could address the use cases using various web technologies.  
+
+We present them here only to foster discussion, and the working group has not settled on any particular one. We are open to feedback to pursuing any of the proposals below. Or, if there is a better alternative solution we have not considered, we would be open to hearing it!
+
+### Screen IDL interface extension
+
+Extend the [Screen interface](https://drafts.csswg.org/cssom-view/#the-screen-interface) with a new operation, inspired by the [Wake Lock API](https://developer.mozilla.org/en-US/docs/Web/API/WakeLockSentinel). Something like:
+
+```webidl
+[SecureContext]
+partial interface Screen {
+  Promise<ScreenBrightnessSentinel> requestBrightnessIncrease();
+};
+
+interface ScreenBrightnessSentinel {
+  Promise<undefined> release();
+};
+```
+
+Proposed usage:
+
+```javascript
+myButton.addEventListener("click", async () => {
+  try {
+    const sentinel = await screen.requestBrightnessIncrease();
+    console.log("The display's brightness level has increased");
+    // …and release it after 5s.
+    window.setTimeout(() => {
+      await sentinel.release();
+    }, 5000);
+  } catch (e) {
+    console.error(e);
+  }
+});
+```
+
+### Screen Wake Lock integration
+
+The idea was to extend the existing Screen Wake Lock API and tie the change in brightness to a WakeLockSentinel. Something like:
+
+``` javascript
+const lock = await navigator.wakeLock.request({ increaseBrightness: true });
+```
+
+This was considered for a long time until the [2021-11-17 DAS WG meeting to discuss the topic](https://www.w3.org/events/meetings/0f623aa1-2026-4366-846b-c2faedda4180).
+
+### navigator.screenBrightness
+
+The API proposed there had a larger surface that allowed reading the brightness value (leading to potential fingerprinting issues). Originally proposed in [WICG issue 17](https://github.com/WICG/proposals/issues/17).
+
+### requestFullscreen() integration
+
+The idea to integrate with `.requestFullscreen()`, to do something like:
+
+``` javascript
+body.requestFullscreen({ increaseBrightness: true });
+```
+
+To take advantage of existing UA privacy mitigations and UX indications that would show to the user that a web application is increasing the brightness, and leaving full screen mode would make it clear that the UA should stop increasing the device's screen's brightness level.
+See [whatwg/fullscreen](https://fullscreen.spec.whatwg.org/) issues [185](https://github.com/whatwg/fullscreen/issues/185) and [198](https://github.com/whatwg/fullscreen/issues/198).
+
+### getUserMedia() integration
```

Comment:
I recall this being put aside for a number of reasons as there's no MediaStream you can get out of screen brightness, and it'd be unclear how to deal with constraint application / failures etc.

I'm not particularly passionate about removing it (or including it) either way, though.

Reply:
I've expanded the text below and mentioned the MediaStream bit from your comment. Please let me know if you feel there's more to add.